src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -23984,5 +23984,1108 @@ class C2 : I1<C2>
                 Diagnostic(ErrorCode.ERR_ValConstraintNotSatisfied, "I1<C2>").WithArguments("I1<T>", "T", "C2").WithLocation(14, 21)
                 );
         }
+
+        [Theory]
+        [CombinatorialData]
+        public void ConsumeAbstractConversionOperator_01([CombinatorialValues("implicit", "explicit")] string op)
+        {
+            string cast = (op == "explicit" ? "(int)" : "");
+
+            var source1 =
+@"
+interface I1<T> where T : I1<T>
+{
+    abstract static " + op + @" operator int(T x);
+
+    static int M02(I1<T> x)
+    {
+        return " + cast + @"x;
+    }
+
+    int M03(I1<T> y)
+    {
+        return " + cast + @"y;
+    }
+}
+
+class Test<T> where T : I1<T>
+{
+    static int MT1(I1<T> a)
+    {
+        return " + cast + @"a;
+    }
+
+    static void MT2()
+    {
+        _ = (System.Linq.Expressions.Expression<System.Func<T, int>>)((T b) => " + cast + @"b);
+    }
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            var error = (op == "explicit" ? ErrorCode.ERR_NoExplicitConv : ErrorCode.ERR_NoImplicitConv);
+
+            compilation1.VerifyDiagnostics(
+                // (8,16): error CS0030: Cannot convert type 'I1<T>' to 'int'
```

Comment:
We might want to consider a better diagnostic experience here. Perhaps let the conversion exist, but make it invalid?

Reply:
> We might want to consider a better diagnostic experience here. Perhaps let the conversion exist, but make it invalid?

The proposed specification doesn't consider any user-defined conversion operators from interface types. If specification changes, things could change.